docs/src/pages/customization/how-to-customize/how-to-customize.md

```diff
@@ -145,8 +145,31 @@ Here are four alternatives; each has its pros and cons.
 
 ### Dynamic CSS
 
+Using the `styled()` utility offers a simple way for adding dynamic styles based on props.
+
 {{"demo": "pages/customization/how-to-customize/DynamicCSS.js", "defaultCodeOpen": false}}
 
+> ⚠️ Note that if you are using TypeScript you will need to update the prop's types of the new component.
+
+```tsx
+import * as React from 'react';
+import { experimentalStyled as styled } from '@material-ui/core/styles';
+import Slider, { SliderProps } from '@material-ui/core/Slider';
+
+type StyledSliderProps = SliderProps & {
+  success?: boolean;
+};
```

Comment:
question: Can we type this as an `interface` instead? [`interface` should be preferred over `type`](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections).

Reply:
Done